packages/flutter_tools/lib/src/host_validator.dart

```diff
@@ -0,0 +1,90 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:http/http.dart' as http;
+
+import 'doctor.dart';
+import 'globals.dart';
+
+/// Verify that the required hosts ( they are stored at [_requiredHosts] ) are reachable
+class HostValidator extends DoctorValidator {
+  HostValidator() : super('Host availability');
+
+  @override
+  String get slowWarning => 'Host availability check is taking a long time...';
+
+  http.Client httpClient = http.Client();
+
+  /// Hosts which are required for flutter
+  /// For flutter build for example
+  List<String> get _requiredHosts {
+    final List<String> hosts = <String>[
+      'https://pub.dev/',
+      'https://cloud.google.com/',
+      'https://maven.google.com/',
+    ];
+    if (platform.isMacOS) {
```

Comment:
Instead of using the values in globals.dart, inject these in the constructor like:

```
HostValidator({
  @required Platform platform,
}) : _platform = platform;

final Platform _platform;
```